src/config.rs

```diff
@@ -546,6 +546,19 @@ fn config_from_env() -> EnvConfig {
     EnvConfig { cache }
 }
 
+fn config_file(env_var: &str, leaf: &str) -> PathBuf {
```

Comment:
Regardless of what we do, we should have a comment describing why we need to do this config-file-search dance.